Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,300 @@
+   also allowing users to install their own Python packages
+2. Provide a `mantid` framework library for users to use in
+   their own projects, e.g mslice, PyRS + other SNS GUIs
+
+## Solutions
+
+Here we describe possible solutions to the problems outlined above.
+The solutions considered in detail are:
+
+* [Conan](#Conan)
+* [Conda](#Conda)
+
+Other solutions not considered:
+
+* Flatpak/Snap - would only work for Linux and still be a large effort.
+* Singularity/Docker - unsure about desktop applications & would not work for macOS.
+
+
+### Conan
+
+[Conan](https://docs.conan.io/en/latest/introduction.html)
+is a decentralized package manager for C/C++.
+It aims to be cross-platform and build system agnostic.
+The servers store packages that can be downloaded directly
+if the required config matches the prebuilt versions.
+Otherwise conan will build packages locally if the requested binaries
+are not available to match the required configuration.
+
+Conan allows for the conan server to be hosted "privately" to facilitate
+sharing binaries across developers on the same project and avoids the overhead
+of each developer rebuilding the dependency set.
+
+Conan would satisfy the requirements 2-4 through the
+[conanfile.txt](https://docs.conan.io/en/latest/reference/conanfile_txt.html)
+or [conanfile.py](https://docs.conan.io/en/latest/reference/conanfile.html).
+
+For user packaging purposes the dependencies would be bundled in with each of
+the user packages on all of the platforms, much like the current approach
+for Windows/macOS.
+User packages would still be created in the appropriate format for that
+platform, e.g. rpm for RHEL/CentOS etc.
+Each package would ship with its own dependencies then the final requirement of
+side-by-side installs will be taken care of automatically.
+QtCreator does this and is able to ship a standalone version with the latest
+Qt version regardless of the system version of Qt.
+
+Another option to consider would be static linking but this may well
+be worth considering separately after the dependency management issue
+is settled.
+
+On Linux this would also have the advantage of supporting additional Linux
+distributions or newer versions of existing distributions almost trivially.
+
+#### CMake
+
+Conan has built in support for
+[CMake](https://docs.conan.io/en/latest/integrations/cmake.html)
+along with a [CMake Wrapper](https://github.com/conan-io/cmake-conan/)
+to enable configuring dependencies completely in CMake.
+
+Developers would continue to work as they do now with their favourite IDEs
+and tools but would need to install Conan first.
+Release packages are [provided](https://github.com/conan-io/conan)
+that have no other dependencies and are easy to install.
+
+#### Python Packages
+
+Conan was designed as a framework to package C/C++ dependencies.
+While it is written in Python itself it does not possess the capability
+to manage Python dependencies. Windows/macOS already ship a bundled Python
```

Comment:
Just to clarify, is it Conan ships Python on these platforms? As it reads that Windows and Mac ship Python